index.bs

```diff
@@ -895,47 +895,49 @@ spec: webidl
       </dt>
       <dd>
         A list of {{MediaDeviceInfo/deviceId}} values for the devices the user
-        has granted access to.
-      </dd>
-      <dt>
-        <a>permission result type</a>
-      </dt>
-      <dd>
-        TODO
+        has made a non-default decision on access to.
       </dd>
       <dt>
         <a>permission query algorithm</a>
       </dt>
       <dd>
-        TODO
+        The permission query algorithm runs the following steps:
+        <ol class="algorithm">
+        <li>
+        If |permissionDesc|.deviceId exists in the <a>extra permission data</a>,
+        set <code><var>status</var>.state</code>
+        to |permissionDesc|'s <a>permission state</a> and terminate these
+        steps.
+        </li>
+        <li>Set <var>global</var> to the {{PermissionDesc}} for the descriptor
+        that results from removing the "deviceId" value.
+        </li>
+        <li>
+        Set <code><var>status</var>.state</code> to <var>global</var>'s
```

Comment:
I don't see a need for an application to get that list, no. If it really wants one, it can call enumerateDevices and then do query() for each of them. I want to keep the API as simple as possible.